src/Sylius/Behat/Context/Api/Shop/ProductVariantContext.php

```diff
@@ -100,4 +100,14 @@ public function iShouldSeeVariantsAreNotDiscounted(ProductVariantInterface ...$v
             Assert::keyNotExists($item, 'appliedPromotions');
         }
     }
+
+    private function findVariant(?ProductVariantInterface $variant): array {
+        $response = $this->client->getLastResponse();
+
+        if($variant !== null && str_contains($response->getContent(), 'hydra:member')) {
```

Comment:
I would be for checking here the `@type` of response if it is `hydra:Collection` or `ProductVariant`, sth like:
```suggestion
        if ($variant !== null && $this->responseChecker->hasValue($response, '@type', 'hydra:Collection')) {
```
but I would be for checking here the `@type` of response if it is `hydra:Collection`